llvm/lib/Transforms/Scalar/FPBuiltinFnSelection.cpp

```diff
@@ -102,8 +103,17 @@ static bool replaceWithLLVMIR(FPBuiltinIntrinsic &BuiltinCall) {
   BuiltinCall.replaceAllUsesWith(Replacement);
   // ConstantFolder may fold original arguments to a constant, meaning we might
   // have no instruction anymore.
-  if (auto *ReplacementI = dyn_cast<Instruction>(Replacement))
+  if (auto *ReplacementI = dyn_cast<Instruction>(Replacement)) {
     ReplacementI->copyFastMathFlags(&BuiltinCall);
+    // Copy accuracy from fp-max-error attribute to fpmath metadata just in case
```

Comment:
Can this not be done when the BuiltinCall is generated by CodeGen? Is there a reason to wait till SYCLLowerIR stage?

This is mostly a nit/curiosity question. Not a blocker.

Thanks

Reply:
AFAIU only entering `CodeGen` we know, what `AltMath` library can offer us in terms of `FPBuiltin` implementations. So at this stage it makes sense to start modifying LLVM IR as we will go over fpbuiltin calls anyway.